MeesGame/Gameplay/Generator/MazeGeneratorMain.cs

```diff
@@ -0,0 +1,148 @@
+﻿using MeesGame;
+using Microsoft.Xna.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace MeesGen
+{
+    partial class MazeGenerator
+    {
+        /// <summary>
+        /// Generates a random maze using backtracking. Use odd dimensions for best effect (even dimensions will be extended).
+        /// </summary>
+        /// <param name="numRows">The number of rows the generated maze should have</param>
+        /// <param name="numCols">The number of columns the generated maze should have</param>
+        /// <param name="difficulty">The difficulty if the generated maze. 1-5</param>
+        /// <returns>The populated TileField</returns>
+        public static TileField GenerateMaze(int numRows = DEFAULT_NUM_ROWS, int numCols = DEFAULT_NUM_COLS, int difficulty = 3)
+        {
+            MazeGenerator mazeGen = new MazeGenerator();
+            mazeGen.InitMazeGen(numRows, numCols, difficulty);
+            mazeGen.MazeGenMain();
+            return mazeGen.Finish();
+        }
+
+        private void MazeGenMain()
+        {
+            bool placeDoors = false; // Only place doors after a key has been placed.
+            bool randomNext = false; // Is set to true if the current path can't be extended
+            Tile tileToAdd;
+            int nodesDone = -1; // Makes it 0 in the first iteration.
+
+            // Add hallways until all tiles are filled (main algorithm).
+            while (nodesToDo.Count != 0)
+            {
+                nodesDone += 1;
+
+                // Continue from the last node, or take a random one
```

Comment:
You could make this a helper function:

    Point current = determineNextNode();

And then later you can do

    nodesToDo.RemoveAt(nodesToDo.count-1);